source/base/parameter_handler.cc

```diff
@@ -703,14 +704,29 @@ void
 ParameterHandler::declare_entry(const std::string &          entry,
                                 const std::string &          default_value,
                                 const Patterns::PatternBase &pattern,
-                                const std::string &          documentation)
+                                const std::string &          documentation,
+                                const bool &                 has_to_be_set)
 {
   entries->put(get_current_full_path(entry) + path_separator + "value",
                default_value);
   entries->put(get_current_full_path(entry) + path_separator + "default_value",
                default_value);
   entries->put(get_current_full_path(entry) + path_separator + "documentation",
                documentation);
+  entries->put(get_current_full_path(entry) + path_separator + "has_to_be_set",
+               has_to_be_set);
+  entries->put(get_current_full_path(entry) + path_separator + "has_been_set",
+               false);
```

Comment:
I like the suggested flexibility! Being able to set it via `declare_entry()` and `add_parameter()` is a good idea!

However, I don't think we don't need to add two parameters to the tree. I would suggest:
```suggestion
  entries->put(get_current_full_path(entry) + path_separator + "has_seen_set",
               !has_to_be_set);
```
or alternately:
```suggestion
  if(has_to_be_set)
    entries->put(get_current_full_path(entry) + path_separator + "has_seen_set",
                 false);
```


Also, I would suggest in this case to eliminate the map (because it only contains duplicated information).

Reply:
The first option means that ``has_been_set`` is set to true if ``has_to_be_set==false``, the second option that only some entries get the property ``has_been_set``?? In both  cases, the logic is not clear to me.